mock/src/transaction.rs

```diff
@@ -190,6 +314,40 @@ impl MockTransaction {
     /// Consumes the mutable ref to the MockTransaction returning the structure
     /// by value.
     pub fn build(&mut self) -> Self {
+        let tx = TransactionRequest::new()
+            .from(self.from.address())
+            .to(self.to.clone().unwrap().address())
+            .nonce(self.nonce)
+            .value(self.value)
+            .data(self.input.clone())
+            .gas(self.gas)
+            .gas_price(self.gas_price);
+
+        let tx_rlp = tx.rlp(self.chain_id.low_u64());
+        let sighash = keccak256(tx_rlp.as_ref()).into();
+
+        match (self.v, self.r, self.s) {
+            (None, None, None) => {
+                // Compute sig params and set them in case we have a wallet as `from` attr.
+                if self.from.is_wallet() && self.hash.is_none() {
+                    let sig = self
+                        .from
+                        .as_wallet()
+                        .with_chain_id(self.chain_id.low_u64())
+                        .sign_hash(sighash, true);
+                    // Set sig parameters
+                    self.sig_data((sig.v, sig.r, sig.s));
+                }
+            }
+            _ => unimplemented!(),
```

Comment:
Why is this unimplemented?  Is it incorrect to build a mock tx while setting the signature parameters?

Reply:
It is incorrect to have just set one of them or two. 
Either all of them are set or none.

Reply:
It is incorrect to have just set one of them or two.
Either all of them are set or none.

Reply:
But this case also takes into account the possibility that the 3 signature parameters are set, doesn't it?
In any case I would add a `panic!()` here with a message explaining that all parameters must be either set or unset.  If I see `unimplemented!()` it makes me think this is a TODO.

Reply:
That makes sense. 
Fixed in b25452b